src/util-logopenfile-redis.c

```diff
@@ -0,0 +1,455 @@
+    }
+#endif
+    SCFree(ctx);
+}
+
+static int SCConfLogReopenRedis(LogFileCtx *log_ctx);
+
+#if HAVE_LIBEVENT
+
+#if HIREDIS_MAJOR == 0 && HIREDIS_MINOR < 11
+
+/** \brief RedisConnectCallback() Closes redis log more
+ *  \param c redis async context
+ */
+void RedisConnectCallback(const redisAsyncContext *c)
+{
+	SCLogInfo("Connected to redis server.");
+}
+#else
+
+/** \brief RedisConnectCallback() Closes redis log more
+ *  \param c redis async context
+ *  \param status status reported by async caller
+ */
+void RedisConnectCallback(const redisAsyncContext *c, int status)
+{
+	SCLogInfo("Connected to redis server. Status [%d]", status);
+}
+
+#endif // HIREDIS_MAJOR == 0 && HIREDIS_MINOR < 11
+
+/** \brief RedisDisconnectCallback() Callback when disconnection from redis happens.
+ *  \param c redis async context
+ *  \param status status reported by async caller
+ */
+void RedisDisconnectCallback(const redisAsyncContext *c, int status)
+{
+	SCLogInfo("Disconnected from redis server. Status [%d]", status);
+}
+
+
+/** \brief SCRedisAsyncCommandCallback() Callback when reply from redis happens.
+ *  \param c redis async context
+ *  \param r redis reply
+ *  \param privvata opaque datq with pointer to LogFileCtx
+ */
+static void SCRedisAsyncCommandCallback (redisAsyncContext *async, void *r, void *privdata)
+{
+    LogFileCtx *file_ctx = privdata;
+    redisReply *reply = r;
+    /* Disconnection or lost reply may have happened */
+    if (reply == NULL) {
+        SCConfLogReopenRedis(file_ctx);
+    }
+}
+
+#endif // HAVE_LIBEVENT
+
+/** \brief SCConfLogReopenRedis() Open or re-opens connection to redis for logging.
+ *  \param log_ctx Log file context allocated by caller
+ */
+static int SCConfLogReopenRedis(LogFileCtx *log_ctx)
+{
+	/* only try to reconnect once per second */
+	if (log_ctx->redis_setup.tried >= time(NULL)) {
+		return -1;
+	}
+	SCLogRedisContext * ctx = log_ctx->redis;
+	const char *redis_server = log_ctx->redis_setup.server;
+	int redis_port = log_ctx->redis_setup.port;
+#if HAVE_LIBEVENT
+    /* ASYNC */
+    if (log_ctx->redis_setup.async) {
+        if (ctx->ev_base != NULL) {
+            event_base_loopbreak(ctx->ev_base);
+        }
+        if (ctx->async != NULL)  {
+            redisAsyncFree(ctx->async);
+        }
+        ctx->async = redisAsyncConnect(redis_server, redis_port);
+        if ( ctx->async != NULL && ctx->async->err) {
+            SCLogError(SC_ERR_SOCKET, "Error connecting to redis server: [%s]", ctx->async->errstr);
+            redisAsyncFree(ctx->async);
+            ctx->async = NULL;
+            log_ctx->redis_setup.tried = time(NULL);
+            return -1;
+        }
+        if (ctx->async != NULL)  {
+            redisLibeventAttach(ctx->async,ctx->ev_base);
+            redisAsyncSetConnectCallback(ctx->async,RedisConnectCallback);
+            redisAsyncSetDisconnectCallback(ctx->async,RedisDisconnectCallback);
+            redisAsyncHandleWrite(ctx->async);
+        }
```

Comment:
The case ctx->async == NULL is not handled but it seems to be a real error case.